webapp/src/http_handlers/submit/mod.rs

```diff
@@ -1,34 +1,66 @@
 mod structs;
 
-use super::super::filework::*;
+use std::path::PathBuf;
 use rocket::serde::json::Json;
+use super::super::filework::*;
 use compiler::data::input_data::{InputData, compiler_type::CompilerType};
 use compiler::handler::run_compilation;
-use std::path::PathBuf;
+use super::super::languages::{lang_info::LangInfo, flags_validator::FlagsValidator};
+
+use rocket::State;
+use super::super::LangsInfo;
 
 #[post("/submit", format = "json", data = "<compilation_json>")]
-pub async fn post_submit(compilation_json: Json<structs::InputData>) 
-    -> Json<structs::OutputData>
+pub async fn post_submit(compilation_json: Json<structs::InputData>,
+    langs_info: &State<LangsInfo>) -> Json<structs::OutputData>
+{
+    let compilation_data = compilation_json.into_inner();
+    if langs_info.contains_key(&compilation_data.lang)
+    {
+        let lang_info = &langs_info[&compilation_data.lang];
+        let mut validator = FlagsValidator::new();
```

Comment:
I don't see the reason why the FlagsValidator is visible to this function.
Is it possible to move validation inside `try_to_compile` or embed validation into the serialization (or conversion to internal representation) of flags?

Reply:
Moving it into `try_to_compile` woud be right, don't know why I didn't do that in the first place.
I see how embeding the entire validation into rocket's data guard could be done by implementing custom `FromRequest` and a `Catcher` for it. I will look into it, but I'm not sure I'd be able to finish it for the demo